src/sdJwt/presentationFrame.ts

```diff
@@ -0,0 +1,93 @@
+import { getByPath, hasByPath, traverseNodes } from '../utils'
+import { Disclosure } from './disclosures'
+import { SdJwtError } from './error'
+import { Hasher } from '../types'
+import { PresentationFrame } from '../types/present'
+import {
+    getDisclosureMap,
+    getPayloadDisclosureMapping
+} from './disclosureMapping'
+
+export const getDisclosuresForPresentationFrame = async <
+    Payload extends Record<string, unknown> = Record<string, unknown>
+>(
+    signedPayload: Payload,
+    presentationFrame: PresentationFrame<Payload>,
+    prettyClaims: Payload,
+    hasher: Hasher,
+    disclosures: Array<Disclosure> = []
+): Promise<Array<Disclosure>> => {
+    const requiredDisclosureDigests = new Set<string>()
+    const disclosureMap = await getDisclosureMap(disclosures, hasher)
+    const payloadDisclosureMapping = getPayloadDisclosureMapping(
+        signedPayload,
+        disclosureMap
+    )
+
+    for (const node of traverseNodes(presentationFrame)) {
+        // We only want to process leaf nodes here
+        if (!node.isLeaf) continue
+
+        if (typeof node.value !== 'boolean') {
+            throw new SdJwtError(
+                `Expected leaf value in presentation frame to be of type boolean, but found ${typeof node.value}`
+            )
+        }
+
+        // If the value is false, it means we don't want to disclose it
+        if (node.value === false) continue
+
+        // FIXME: we should check whether the presentation frame matches the type of the payload. So if the presentation frame has an array as value, but the pretty payload
```

Comment:
> So if the presentation frame has an array as value, but the pretty payload

What does this mean?

Reply:
Lets say the payload is

```
{
  "array": [1, 2]
}
```

But the presentation frame is:

```
{
  "array": { "firstKey" }
}
```

I'm nit really sure what will happen atm. Might be that it already handles this, but not sure